ngraph/python/src/pyngraph/passes/manager.cpp

```diff
@@ -18,13 +18,45 @@
 #include <pybind11/stl.h>
 
 #include "ngraph/pass/manager.hpp"
+#include "ngraph/pass/pass.hpp"
+
+#include "ngraph/pass/constant_folding.hpp"
+#include "ngraph/pass/validate.hpp"
+
 #include "pyngraph/passes/manager.hpp"
 
 namespace py = pybind11;
 
+namespace
+{
+    class ManagerWrapper : public ngraph::pass::Manager
+    {
+    public:
+        ManagerWrapper() {}
+        ~ManagerWrapper() {}
+        void register_pass(std::string pass_name)
+        {
+            if (pass_name == "ConstantFolding")
+                push_pass<ngraph::pass::ConstantFolding>();
+
+            if (m_per_pass_validation)
+            {
+                push_pass<ngraph::pass::Validate>();
+            }
+            return;
+        }
+    };
+}
+
 void regclass_pyngraph_passes_Manager(py::module m)
 {
-    py::class_<ngraph::pass::Manager, std::shared_ptr<ngraph::pass::Manager>> manager(m, "Manager");
-    manager.doc() = "ngraph.impl.pass.Manager wraps ngraph::pass::Manager";
-    manager.def("run_passes", &ngraph::pass::Manager::run_passes);
+    py::class_<ManagerWrapper> manager(m, "Manager");
```

Comment:
I think we should nest this module in a `pass` module to represent the `ngraph.pass.Manager` hierarchy.

Reply:
There is already a nested submodule called `passes` and it's suitable since `pass` is reserved keyword for Python.